src/providers/ipa/ipa_s2n_exop.c

```diff
@@ -1750,6 +1750,106 @@ static errno_t get_group_dn_list(TALLOC_CTX *mem_ctx,
+    case BE_REQ_USER:
+        if (req_input->type == REQ_INP_NAME) {
+            name_is_upn = strchr(req_input->inp.name, '@') == NULL ? false
+                                                                   : true;
+            /* Expand to fully-qualified internal name */
+            if (!name_is_upn) {
+                fq_name = sss_create_internal_fqname(mem_ctx,
+                                                     req_input->inp.name,
+                                                     domain->name);
+                if (fq_name == NULL) {
+                    DEBUG(SSSDBG_OP_FAILURE,
+                          "sss_create_internal_fqname failed.\n");
+                    ret = ENOMEM;
+                    goto done;
+                }
+            }
+            ret = users_get_handle_no_user(mem_ctx, domain, BE_FILTER_NAME,
+                                           fq_name != NULL ? fq_name
+                                                          : req_input->inp.name,
+                                           name_is_upn);
+        } else if (req_input->type == REQ_INP_ID) {
+            ret = users_get_handle_no_user(mem_ctx, domain, BE_FILTER_IDNUM,
+                                           id_str, false);
+        } else {
+            DEBUG(SSSDBG_OP_FAILURE, "Unexpected input type [%d].\n",
+                                      req_input->type);
+            ret = EINVAL;
+            goto done;
+        }
+        if (ret != EOK || entry_type == BE_REQ_USER) {
+            break;
+        }
+        /* Fallthough if BE_REQ_USER_AND_GROUP */
+    case BE_REQ_GROUP:
+        if (req_input->type == REQ_INP_NAME) {
+            /* Expand to fully-qualified internal name */
+            fq_name = sss_create_internal_fqname(mem_ctx,
+                                                 req_input->inp.name,
+                                                 domain->name);
+            if (fq_name == NULL) {
+                DEBUG(SSSDBG_OP_FAILURE,
+                      "sss_create_internal_fqname failed.\n");
+                ret = ENOMEM;
+                goto done;
+            }
+            ret = groups_get_handle_no_group(mem_ctx, domain, BE_FILTER_NAME,
+                                             fq_name);
+        } else if (req_input->type == REQ_INP_ID) {
+            ret = groups_get_handle_no_group(mem_ctx, domain,BE_FILTER_IDNUM,
+                                             id_str);
+        } else {
+            DEBUG(SSSDBG_OP_FAILURE, "Unexpected input type [%d].\n",
+                                      req_input->type);
+            ret = EINVAL;
+            goto done;
+        }
+        break;
+    case BE_REQ_BY_SECID:
+        ret = EOK;
+        break;
+    case BE_REQ_BY_CERT:
+        ret = EOK;
+        break;
+    default:
+        DEBUG(SSSDBG_OP_FAILURE, "Unexpected entry type [%d].\n", entry_type);
+        ret = EINVAL;
+    }
+
+done:
+    if (ret != EOK) {
```

Comment:
I think ENOENT should be completely ignored here. During my testing, when I try to request an object that was recently removed, the first request is by name and removes the object. But then comes the same request from the responder for UPN, does not find the object either and tries to remove it again which produces some loud debug messages.